src/shared/components/Input.tsx

```diff
@@ -11,18 +30,9 @@ export default function SharedInput({
   disabled = false,
   rightComponent = null,
   style = {},
-}: {
-  label: string
-  type?: "text" | "number"
-  disabled?: boolean
-  value: string
-  placeholder?: string
-  onChange?: (value: string) => void
-  validate?: (value: string) => { value: unknown } | { error: string }
-  rightComponent?: React.ReactNode
-  style?: React.CSSProperties & Record<string, string>
-}) {
+}: SharedInputProps) {
   const [error, setError] = useState("")
+  const isTypeNumber = useMemo(() => type === "number", [type])
```

Comment:
is useMemo doing anything? both "type" is a string and "isTypeNumber" is a boolean - they shouldn't cause rerenders or trigger changes without actually changing values, right?